src/bootstrap/src/utils/tests/mod.rs

```diff
@@ -117,4 +125,19 @@ impl ConfigBuilder {
 
         Config::parse(Flags::parse(&self.args))
     }
+
+    pub fn create_config_without_ci_llvm_override(mut self) -> Config {
```

Comment:
It's not needed to duplicate the whole logic there. Instead, store an `bool` in the builder, which will determine whether we disable `download-ci-llvm` or not. It will be set to `true` by default, but it will be overrideable in certain tests that need a different value (e.g. `.no_override_download_ci_llvm()`). And then in `create_config` only add the option if it's set to `true`.